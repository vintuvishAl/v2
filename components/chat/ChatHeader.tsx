import { Ionicons } from "@expo/vector-icons";
import React, { useCallback } from "react";
import { Text, TouchableOpacity, View } from "react-native";
import { useHapticFeedback } from "../../hooks/useHapticFeedback";

interface ChatHeaderProps {
  onMenuPress: () => void;
  onSearchPress: () => void;
  onNewChatPress: () => void;
  useAgent: boolean;
  onToggleAgent: (enabled: boolean) => void;
}

export const ChatHeader: React.FC<ChatHeaderProps> = React.memo(({
  onMenuPress,
  onSearchPress,
  onNewChatPress,
  useAgent,
  onToggleAgent,
}) => {
  const { onButtonPress, onImportantAction } = useHapticFeedback();

  const handleMenuPress = useCallback(() => {
    // Add haptic feedback for menu button
    onButtonPress();
    onMenuPress();
  }, [onButtonPress, onMenuPress]);

  const handleNewChatPress = useCallback(() => {
    // Add haptic feedback for new chat button
    onImportantAction();
    onNewChatPress();
  }, [onImportantAction, onNewChatPress]);

  return (
    <View className="flex-row justify-between items-center px-6 py-4 bg-app-dark-background">
      <View className="flex-row items-center">        <TouchableOpacity testID="all-chats-button" className="p-2 " onPress={handleMenuPress}>
          <Ionicons
            name="menu"
            size={30}
            className="text-app-dark-icon"
            color={"#1cb8ff"}
          />
        </TouchableOpacity>
        <TouchableOpacity
          testID="add-button"
          className="p-2"
          onPress={handleNewChatPress}
        >
          <Ionicons
            name="add"
            size={30}
            className="text-app-dark-icon"
            color={"#1cb8ff"}
          />
        </TouchableOpacity>        {/* <TouchableOpacity testID="search-button" className="p-2" onPress={onSearchPress}>
          <Ionicons
            name="search"
            size={30}
            className="text-app-dark-icon"
            color={"#1cb8ff"}
          />
        </TouchableOpacity> */}
        
      </View>

      <View className="flex-row items-center space-x-2">
        <Text className="text-app-dark-text text-xl font-bold">
          V2 Chat
        </Text>      </View>
    </View>
  );
});

ChatHeader.displayName = 'ChatHeader';
