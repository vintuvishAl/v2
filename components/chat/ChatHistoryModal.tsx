import { Ionicons } from "@expo/vector-icons";
import { FlashList } from "@shopify/flash-list";
import React, { useCallback } from "react";
import { Modal, Text, TouchableOpacity, View } from "react-native";
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withSpring,
  withTiming
} from 'react-native-reanimated';
import type { Id } from "../../convex/_generated/dataModel";
import { useHapticFeedback } from "../../hooks/useHapticFeedback";

interface ChatHistoryItem {
  _id: Id<"chats">;
  title: string;
  createdAt: number;
  updatedAt?: number;
}

interface ChatHistoryModalProps {
  visible: boolean;
  onClose: () => void;
  chatHistory: ChatHistoryItem[];
  onChatSelect: (chatId: Id<"chats">) => void;
  onDeleteChat: (chatId: Id<"chats">) => void;
  onNewChat: () => void;
  loading?: boolean;
}

export const ChatHistoryModal: React.FC<ChatHistoryModalProps> = ({
  visible,
  onClose,
  chatHistory,
  onChatSelect,
  onDeleteChat,
  onNewChat,
  loading = false,
}) => {
  const { onButtonPress, onImportantAction, onCriticalAction } = useHapticFeedback();
  
  // Optimized date formatting with memoization
  const formatDate = useCallback((timestamp: number) => {
    const date = new Date(timestamp);
    const now = new Date();
    const diffInHours = (now.getTime() - date.getTime()) / (1000 * 60 * 60);
    
    if (diffInHours < 24) {
      return date.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });
    } else if (diffInHours < 24 * 7) {
      return date.toLocaleDateString([], { weekday: 'short', hour: '2-digit', minute: '2-digit' });
    } else {
      return date.toLocaleDateString([], { month: 'short', day: 'numeric' });
    }
  }, []);

  // Animated values for smooth modal transitions
  const translateY = useSharedValue(1000);
  const opacity = useSharedValue(0);

  // Update animation when modal visibility changes
  React.useEffect(() => {
    if (visible) {
      translateY.value = withSpring(0, { damping: 20, stiffness: 300 });
      opacity.value = withTiming(1, { duration: 200 });
    } else {
      translateY.value = withTiming(1000, { duration: 200 });
      opacity.value = withTiming(0, { duration: 200 });
    }
  }, [visible, translateY, opacity]);

  // Animated styles
  const backgroundStyle = useAnimatedStyle(() => ({
    opacity: opacity.value,
  }));

  const modalStyle = useAnimatedStyle(() => ({
    transform: [{ translateY: translateY.value }],
  }));

  const handleChatPress = useCallback((chatId: Id<"chats">) => {
    onButtonPress();
    onChatSelect(chatId);
    onClose();
  }, [onButtonPress, onChatSelect, onClose]);

  const handleNewChat = useCallback(() => {
    onImportantAction();
    onNewChat();
    onClose();
  }, [onImportantAction, onNewChat, onClose]);

  const handleDeleteChat = useCallback((chatId: Id<"chats">) => {
    onCriticalAction();
    onDeleteChat(chatId);
  }, [onCriticalAction, onDeleteChat]);

  const handleClose = useCallback(() => {
    onButtonPress();
    onClose();
  }, [onButtonPress, onClose]);

  // Memoized render function for chat items to prevent unnecessary re-renders
  const renderChatItem = useCallback(({ item }: { item: ChatHistoryItem }) => (
    <View className="flex-row items-center px-4 py-3 border-b border-gray-700">
      <TouchableOpacity
        className="flex-1 mr-3"
        onPress={() => handleChatPress(item._id)}
      >
        <Text className="text-white text-base font-medium mb-1" numberOfLines={1}>
          {item.title}
        </Text>
        <Text className="text-gray-400 text-sm">
          {formatDate(item.updatedAt || item.createdAt)}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        className="p-2"
        onPress={() => handleDeleteChat(item._id)}
      >
        <Ionicons name="trash-outline" size={18} color="#ef4444" />
      </TouchableOpacity>
    </View>
  ), [handleChatPress, handleDeleteChat, formatDate]);

  // Memoized key extractor
  const keyExtractor = useCallback((item: ChatHistoryItem) => item._id, []);

  return (
    <Modal
      visible={visible}
      transparent={true}
      animationType="none"
      onRequestClose={onClose}
    >
      <Animated.View style={[{ flex: 1, backgroundColor: 'rgba(0,0,0,0.5)' }, backgroundStyle]}>
        <Animated.View style={[{ flex: 1, backgroundColor: '#111827' }, modalStyle]}>
          <View className="flex-row justify-between items-center px-4 py-4 border-b border-gray-700">
            <Text className="text-white text-xl font-semibold">Chat History</Text>
            <TouchableOpacity onPress={handleClose}>
              <Ionicons name="close" size={24} color="white" />
            </TouchableOpacity>
          </View>
          <TouchableOpacity
            className="flex-row items-center px-4 py-4 bg-blue-600 mx-4 mt-4 rounded-lg"
            onPress={handleNewChat}
          >
            <Ionicons name="add" size={20} color="white" />
            <Text className="text-white text-base font-medium ml-2">Start New Chat</Text>
          </TouchableOpacity>
          <View className="flex-1 mt-4">
            {loading ? (
              <View className="flex-1 justify-center items-center">
                <Text className="text-gray-400">Loading chat history...</Text>
              </View>
            ) : chatHistory.length === 0 ? (
              <View className="flex-1 justify-center items-center px-4">
                <Ionicons name="chatbubbles-outline" size={64} color="#6b7280" />
                <Text className="text-gray-400 text-lg mt-4 mb-2">No chat history</Text>
                <Text className="text-gray-500 text-center">
                  Start a conversation to see your chat history here
                </Text>
              </View>
            ) : (              <FlashList
                data={chatHistory}
                renderItem={renderChatItem}
                keyExtractor={keyExtractor}
                showsVerticalScrollIndicator={false}
                estimatedItemSize={70}
                removeClippedSubviews={true}
                getItemType={() => 'chat-item'}
              />
            )}
          </View>
        </Animated.View>
      </Animated.View>
    </Modal>
  );
};
