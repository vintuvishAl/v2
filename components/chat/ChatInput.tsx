import { Ionicons } from "@expo/vector-icons";
import React, { useCallback } from "react";
import {
  Keyboard,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from "react-native";
import type { ModelType } from "../../hooks/useChat";
import { useHapticFeedback } from "../../hooks/useHapticFeedback";

export const modelOptions = [
  { id: "gemini-2.5-flash", name: "Gemini 2.5 Flash", provider: "Google" },
  { id: "gemini-2.5-pro", name: "Gemini 2.5 Pro", provider: "Google" },
  { id: "gpt-4o-mini", name: "GPT-4o Mini", provider: "OpenAI" },
  { id: "gpt-4o", name: "GPT-4o", provider: "OpenAI" },
] as const;

interface ChatInputProps {
  inputText: string;
  onInputChange: (text: string) => void;
  onSendMessage: () => void;
  onModelSelectorPress: () => void;
  onGlobePress: () => void;
  onAttachPress: () => void;
  selectedModel: ModelType;
  isCreatingChat: boolean;
  useAgent?: boolean;
}

export const ChatInput: React.FC<ChatInputProps> = React.memo(({
  inputText,
  onInputChange,
  onSendMessage,
  onModelSelectorPress,
  onGlobePress,
  onAttachPress,
  selectedModel,
  isCreatingChat,
  useAgent = false,
}) => {
  const { onImportantAction, onButtonPress } = useHapticFeedback();

  const handleSendMessage = useCallback(() => {
    if (inputText.trim() && !isCreatingChat) {
      // Add haptic feedback for sending messages
      onImportantAction();
      Keyboard.dismiss();
      onSendMessage();
    }
  }, [inputText, isCreatingChat, onImportantAction, onSendMessage]);

  const handleModelSelectorPress = useCallback(() => {
    // Add haptic feedback for model selection
    onButtonPress();
    onModelSelectorPress();
  }, [onButtonPress, onModelSelectorPress]);

  // Memoize model name to prevent recalculation
  const selectedModelName = React.useMemo(() => {
    return modelOptions.find((m) => m.id === selectedModel)?.name;
  }, [selectedModel]);

  return (
    <View className="bg-app-dark-background px-3 py-0">
      {/* Input Field - Separate section */}
      <View className="border-4 border-app-dark-border border-b-0 rounded-lg rounded-b-none   ">
        <View className="bg-app-dark-chat-bg  px-6 py-2  ">
          {" "}
          <TextInput
            className="text-app-dark-text text-base min-h-6"
            placeholder="Type your message here..."
            placeholderTextColor="#9ca3af"
            value={inputText}
            onChangeText={onInputChange}
            onSubmitEditing={handleSendMessage}
            blurOnSubmit={false}
            returnKeyType="send"            multiline
            testID="message-input"
          />
        </View>
        {/* Model Info Bar - Separate section */}
        <View className="flex-row items-center justify-between bg-app-dark-chat-bg  px-6 py-2 ">
          <View className="flex-row items-center space-x-4 gap-2">
            {/* Agent Mode Indicator */}
            {useAgent && (
              <View className="flex-row items-center space-x-1 bg-blue-600 px-2 py-1 rounded">
                <Ionicons name="calculator" size={14} color="white" />
                <Text className="text-white text-xs font-medium">Agent</Text>
              </View>
            )}            {/* Model Selection Button */}            <TouchableOpacity
              className="flex-row items-center space-x-1 bg-app-dark-border px-3 py-2 rounded-lg"
              testID="model-selector"
              onPress={handleModelSelectorPress}
            >
              <Text className="text-app-dark-text text-sm font-medium">
                {selectedModelName}
              </Text>
              <Ionicons
                name="chevron-down"
                size={16}
                className="text-app-dark-icon"
                color={"#6b6b6b"}
              />
            </TouchableOpacity>

            {/* <TouchableOpacity testID="globe-button" onPress={handleGlobePress}>
              <Ionicons
                name="globe-outline"
                size={20}
                className="text-app-dark-icon"
                color={"#6b6b6b"}
              />
            </TouchableOpacity>
            
           
            <TouchableOpacity testID="attach-button" onPress={handleAttachPress}>
              <Ionicons
                name="attach"
                size={20}
                className="text-app-dark-icon"
                color={"#6b6b6b"}
              />
            </TouchableOpacity> */}
          </View>
          {/* Send Button */}
          <TouchableOpacity
            className={`px-4 py-3 rounded-xl ${
              inputText.trim() && !isCreatingChat
                ? "bg-app-dark-tint"
                : "bg-app-dark-border"
            }`}
            onPress={handleSendMessage}
            disabled={!inputText.trim() || isCreatingChat}
            testID="send-button"
          >
            {isCreatingChat ? (
              <Ionicons
                name="hourglass"
                size={20}
                className="text-app-dark-text"
                color={"#6b6b6b"}
              />
            ) : (
              <Ionicons
                name="arrow-up"
                size={20}
                className="text-app-dark-text"
                color={"#1f2937"}
              />
            )}
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
});

ChatInput.displayName = 'ChatInput';
