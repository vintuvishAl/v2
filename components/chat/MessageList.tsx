import React from "react";
import { Text, View } from "react-native";
import Markdown from "react-native-markdown-display";
import type { Message } from "../../hooks/useChat";

interface MessageListProps {
  messages: Message[];
}

export const MessageList: React.FC<MessageListProps> = ({ messages }) => {  // Deduplicate messages based on content and user type to prevent double rendering
  const deduplicatedMessages = React.useMemo(() => {
    const result: Message[] = [];
    const aiMessages: Message[] = [];
    
    // Separate user and AI messages
    messages.forEach(message => {
      if (message.isUser) {
        result.push(message);
      } else {
        aiMessages.push(message);
      }
    });
    
    // For AI messages, remove duplicates and partial duplicates
    const processedAIMessages: Message[] = [];
    aiMessages.forEach(message => {
      // Check if this message is a subset of any existing message
      const isSubsetOfExisting = processedAIMessages.some(existing => 
        existing.text.includes(message.text) && existing.text.length > message.text.length
      );
      
      if (!isSubsetOfExisting) {
        // Remove any existing messages that are subsets of this message
        const filteredExisting = processedAIMessages.filter(existing => 
          !(message.text.includes(existing.text) && message.text.length > existing.text.length)
        );
        
        // Check for exact duplicates in the filtered list
        const isDuplicate = filteredExisting.some(existing => 
          existing.text.trim() === message.text.trim()
        );
        
        if (!isDuplicate) {
          processedAIMessages.length = 0; // Clear array
          processedAIMessages.push(...filteredExisting, message);
        }
      }
    });
    
    // Combine user messages with deduplicated AI messages, sorted by timestamp
    const combined = [...result, ...processedAIMessages];
    return combined.sort((a, b) => a.timestamp.getTime() - b.timestamp.getTime());
  }, [messages]);
  const markdownStyles = {
    body: {
      color: '#f3f4f6', // text-app-dark-text color
      fontSize: 16,
    },
    code_inline: {
      backgroundColor: '#374151',
      color: '#e5e7eb',
      paddingHorizontal: 4,
      paddingVertical: 2,
      borderRadius: 4,
    },
    code_block: {
      backgroundColor: '#374151',
      color: '#e5e7eb',
      padding: 12,
      borderRadius: 8,
      marginVertical: 8,
    },
    blockquote: {
      backgroundColor: '#1f2937',
      borderLeftColor: '#6b7280',
      borderLeftWidth: 4,
      paddingLeft: 12,
      paddingVertical: 8,
      marginVertical: 8,
    },
    heading1: {
      color: '#f3f4f6',
      fontSize: 24,
      fontWeight: 'bold' as const,
      marginVertical: 8,
    },
    heading2: {
      color: '#f3f4f6',
      fontSize: 20,
      fontWeight: 'bold' as const,
      marginVertical: 6,
    },
    heading3: {
      color: '#f3f4f6',
      fontSize: 18,
      fontWeight: 'bold' as const,
      marginVertical: 4,
    },
    list_item: {
      color: '#f3f4f6',
      marginVertical: 2,
    },
    strong: {
      color: '#f3f4f6',
      fontWeight: 'bold' as const,
    },
    em: {
      color: '#f3f4f6',
      fontStyle: 'italic' as const,
    },
  };
  return (
    <View className="py-4">
      {deduplicatedMessages.map((message) => (        <View
          key={message.id}
          className={`mb-4 ${
            message.isUser ? "items-end" : "items-start"
          }`}
          testID={`message-${message.id}`}
        >
          <View
            className={`px-4 py-2 rounded-lg ${
              message.isUser
                ? "bg-app-dark-user-message ml-auto max-w-xs lg:max-w-md"
                : "bg-app-dark-ai-message mr-auto w-full"
            }`}
          >
            {message.isUser ? (
              <Text className="text-app-dark-text">{message.text}</Text>
            ) : (
              <Markdown style={markdownStyles}>
                {message.text}
              </Markdown>
            )}
          </View>
        </View>
      ))}
    </View>
  );
};
