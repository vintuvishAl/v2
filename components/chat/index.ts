export { ChatHeader } from "./ChatHeader";
export { ChatHistoryModal } from "./ChatHistoryModal";
export { ChatInput } from "./ChatInput";
export { MessageList } from "./MessageList";
export { ModelSelectorModal } from "./ModelSelectorModal";
export { WelcomeScreen } from "./WelcomeScreen";

