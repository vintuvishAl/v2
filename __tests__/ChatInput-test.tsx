import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';

import { ChatInput, modelOptions } from '@/components/chat/ChatInput';

describe('<ChatInput />', () => {
  const mockOnInputChange = jest.fn();
  const mockOnSendMessage = jest.fn();
  const mockOnModelSelectorPress = jest.fn();
  const mockOnGlobePress = jest.fn();
  const mockOnAttachPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const defaultProps = {
    inputText: '',
    onInputChange: mockOnInputChange,
    onSendMessage: mockOnSendMessage,
    onModelSelectorPress: mockOnModelSelectorPress,
    onGlobePress: mockOnGlobePress,
    onAttachPress: mockOnAttachPress,
    selectedModel: 'gemini-2.5-flash' as const,
    isCreatingChat: false,
    useAgent: false,
  };

  test('renders text input with placeholder', () => {
    const { getByPlaceholderText } = render(<ChatInput {...defaultProps} />);
    expect(getByPlaceholderText('Type your message here...')).toBeTruthy();
  });

  test('renders selected model name', () => {
    const { getByText } = render(<ChatInput {...defaultProps} />);
    const expectedModelName = modelOptions.find(m => m.id === 'gemini-2.5-flash')?.name;
    expect(getByText(expectedModelName!)).toBeTruthy();
  });

  test('renders send button', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} />);
    expect(getByTestId('send-button')).toBeTruthy();
  });

  test('renders model selector button', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} />);
    expect(getByTestId('model-selector')).toBeTruthy();
  });

  test('calls onInputChange when typing', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} />);
    const input = getByTestId('message-input');
    
    fireEvent.changeText(input, 'Hello world');
    expect(mockOnInputChange).toHaveBeenCalledWith('Hello world');
  });

  test('calls onSendMessage when send button is pressed', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} inputText="Hello" />);
    const sendButton = getByTestId('send-button');
    
    fireEvent.press(sendButton);
    expect(mockOnSendMessage).toHaveBeenCalledTimes(1);
  });

  test('calls onModelSelectorPress when model selector is pressed', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} />);
    const modelSelector = getByTestId('model-selector');
    
    fireEvent.press(modelSelector);
    expect(mockOnModelSelectorPress).toHaveBeenCalledTimes(1);
  });

  test('send button is disabled when input is empty', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} inputText="" />);
    const sendButton = getByTestId('send-button');
    
    expect(sendButton.props.accessibilityState.disabled).toBeTruthy();
  });

  test('send button is enabled when input has text', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} inputText="Hello" />);
    const sendButton = getByTestId('send-button');
    
    expect(sendButton.props.accessibilityState.disabled).toBeFalsy();
  });

  test('send button is disabled when creating chat', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} inputText="Hello" isCreatingChat={true} />);
    const sendButton = getByTestId('send-button');
    
    expect(sendButton.props.accessibilityState.disabled).toBeTruthy();
  });

  test('shows agent indicator when useAgent is true', () => {
    const { getByText } = render(<ChatInput {...defaultProps} useAgent={true} />);
    expect(getByText('Agent')).toBeTruthy();
  });

  test('does not show agent indicator when useAgent is false', () => {
    const { queryByText } = render(<ChatInput {...defaultProps} useAgent={false} />);
    expect(queryByText('Agent')).toBeNull();
  });

  test('displays correct model name for different models', () => {
    const { getByText } = render(<ChatInput {...defaultProps} selectedModel="gpt-4o" />);
    const expectedModelName = modelOptions.find(m => m.id === 'gpt-4o')?.name;
    expect(getByText(expectedModelName!)).toBeTruthy();
  });

  test('shows loading icon when creating chat', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} isCreatingChat={true} />);
    const sendButton = getByTestId('send-button');
    
    // The button should contain the hourglass icon when creating chat
    expect(sendButton).toBeTruthy();
  });

  test('calls onSendMessage on submit editing', () => {
    const { getByTestId } = render(<ChatInput {...defaultProps} inputText="Hello" />);
    const input = getByTestId('message-input');
    
    fireEvent(input, 'submitEditing');
    expect(mockOnSendMessage).toHaveBeenCalledTimes(1);
  });

  test('matches snapshot', () => {
    const tree = render(<ChatInput {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('matches snapshot with agent mode', () => {
    const tree = render(<ChatInput {...defaultProps} useAgent={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('matches snapshot while creating chat', () => {
    const tree = render(<ChatInput {...defaultProps} isCreatingChat={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
