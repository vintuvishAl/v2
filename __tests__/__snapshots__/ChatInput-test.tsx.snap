// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatInput /> matches snapshot 1`] = `
<View
  className="bg-app-dark-background px-3 py-0"
>
  <View
    className="border-4 border-app-dark-border border-b-0 rounded-lg rounded-b-none   "
  >
    <View
      className="bg-app-dark-chat-bg  px-6 py-2  "
    >
       
      <TextInput
        blurOnSubmit={false}
        className="text-app-dark-text text-base min-h-6"
        multiline={true}
        onChangeText={[MockFunction]}
        onSubmitEditing={[Function]}
        placeholder="Type your message here..."
        placeholderTextColor="#9ca3af"
        returnKeyType="send"
        testID="message-input"
        value=""
      />
    </View>
    <View
      className="flex-row items-center justify-between bg-app-dark-chat-bg  px-6 py-2 "
    >
      <View
        className="flex-row items-center space-x-4 gap-2"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="model-selector"
        >
          <Text
            className="text-app-dark-text text-sm font-medium"
          >
            Gemini 2.5 Flash
          </Text>
          <Text
            allowFontScaling={false}
            className="text-app-dark-icon"
            selectable={false}
            style={
              [
                {
                  "color": "#6b6b6b",
                  "fontSize": 16,
                },
                undefined,
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="send-button"
      >
        <Text
          allowFontScaling={false}
          className="text-app-dark-text"
          selectable={false}
          style={
            [
              {
                "color": "#1f2937",
                "fontSize": 20,
              },
              undefined,
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ChatInput /> matches snapshot while creating chat 1`] = `
<View
  className="bg-app-dark-background px-3 py-0"
>
  <View
    className="border-4 border-app-dark-border border-b-0 rounded-lg rounded-b-none   "
  >
    <View
      className="bg-app-dark-chat-bg  px-6 py-2  "
    >
       
      <TextInput
        blurOnSubmit={false}
        className="text-app-dark-text text-base min-h-6"
        multiline={true}
        onChangeText={[MockFunction]}
        onSubmitEditing={[Function]}
        placeholder="Type your message here..."
        placeholderTextColor="#9ca3af"
        returnKeyType="send"
        testID="message-input"
        value=""
      />
    </View>
    <View
      className="flex-row items-center justify-between bg-app-dark-chat-bg  px-6 py-2 "
    >
      <View
        className="flex-row items-center space-x-4 gap-2"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="model-selector"
        >
          <Text
            className="text-app-dark-text text-sm font-medium"
          >
            Gemini 2.5 Flash
          </Text>
          <Text
            allowFontScaling={false}
            className="text-app-dark-icon"
            selectable={false}
            style={
              [
                {
                  "color": "#6b6b6b",
                  "fontSize": 16,
                },
                undefined,
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="send-button"
      >
        <Text
          allowFontScaling={false}
          className="text-app-dark-text"
          selectable={false}
          style={
            [
              {
                "color": "#6b6b6b",
                "fontSize": 20,
              },
              undefined,
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ChatInput /> matches snapshot with agent mode 1`] = `
<View
  className="bg-app-dark-background px-3 py-0"
>
  <View
    className="border-4 border-app-dark-border border-b-0 rounded-lg rounded-b-none   "
  >
    <View
      className="bg-app-dark-chat-bg  px-6 py-2  "
    >
       
      <TextInput
        blurOnSubmit={false}
        className="text-app-dark-text text-base min-h-6"
        multiline={true}
        onChangeText={[MockFunction]}
        onSubmitEditing={[Function]}
        placeholder="Type your message here..."
        placeholderTextColor="#9ca3af"
        returnKeyType="send"
        testID="message-input"
        value=""
      />
    </View>
    <View
      className="flex-row items-center justify-between bg-app-dark-chat-bg  px-6 py-2 "
    >
      <View
        className="flex-row items-center space-x-4 gap-2"
      >
        <View
          className="flex-row items-center space-x-1 bg-blue-600 px-2 py-1 rounded"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "white",
                  "fontSize": 14,
                },
                undefined,
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            className="text-white text-xs font-medium"
          >
            Agent
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="model-selector"
        >
          <Text
            className="text-app-dark-text text-sm font-medium"
          >
            Gemini 2.5 Flash
          </Text>
          <Text
            allowFontScaling={false}
            className="text-app-dark-icon"
            selectable={false}
            style={
              [
                {
                  "color": "#6b6b6b",
                  "fontSize": 16,
                },
                undefined,
                {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="send-button"
      >
        <Text
          allowFontScaling={false}
          className="text-app-dark-text"
          selectable={false}
          style={
            [
              {
                "color": "#1f2937",
                "fontSize": 20,
              },
              undefined,
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
